sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/WcfNonVoidOneWay.vb

```diff
@@ -0,0 +1,34 @@
+﻿Imports System
+Imports System.Collections.Generic
+Imports System.ServiceModel
+
+Namespace Tests.Diagnostics
+    Public Class MyAttribute
+        Inherits Attribute
+
+        Public Property isOneWay As Boolean
+    End Class
+
+    <ServiceContract>
+    Interface IMyService1
+        <OperationContract>
+        Function MyServiceMethod() As Integer
+
+        <OperationContract(IsOneWay:=True)>
+        Function MyServiceMethod2() As Integer ' Noncompliant {{This method can't return any values because it is marked as one-way operation.}}
+'                                      ^^^^^^^
+
+        <OperationContract(IsOneWay:=False)>
+        Function MyServiceMethod3() As Integer
+
+        <OperationContract(IsTerminating:=True)>
+        <My(IsOneWay:=True)>
+        Function MyServiceMethod4() As Integer
+
+        <OperationContract(IsOneWay:="mistake")>
```

Comment:
could also add a test with `<OperationContract(Action:="Action", IsOneWay:="mistake")>` (where IsOneWay is not the first property in the list)